server/src/main/java/com/cloud/consoleproxy/ConsoleProxyManagerImpl.java

```diff
@@ -401,9 +401,10 @@ public ConsoleProxyVO doAssignProxy(long dataCenterId, long vmId) {
             return null;
         }
 
-        if (vm != null && vm.getState() != State.Starting && vm.getState() != State.Running) {
+        if (vm != null && vm.getState() != State.Starting && vm.getState() != State.Running
+                && vm.getState() != State.Stopping && vm.getState() != State.Migrating) {
```

Comment:
:) a lot of getState()s there

Reply:
@DaanHoogland not very much , only 4 :-) 
will pay attention in the future.

Reply:
NP, code can be such a joy.